modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v4_2_1/test/WorkflowContextUpgradeProcessTest.java

```diff
@@ -0,0 +1,161 @@
+ */
+@RunWith(Arquillian.class)
+public class WorkflowContextUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		WorkflowDefinitionLinkLocalServiceUtil.addWorkflowDefinitionLink(
+			TestPropsValues.getUserId(), TestPropsValues.getCompanyId(),
+			TestPropsValues.getGroupId(), "com.liferay.blogs.model.BlogsEntry",
+			0, 0, "Single Approver", 1);
+
+		Group group = GroupTestUtil.addGroup();
+
+		_blogsEntryLocalService.addEntry(
+			TestPropsValues.getUserId(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(),
+			ServiceContextTestUtil.getServiceContext(group.getGroupId()));
+
+		ServiceContext serviceContext = new ServiceContext();
+
+		serviceContext.setCompanyId(TestPropsValues.getCompanyId());
+		serviceContext.setUserId(TestPropsValues.getUserId());
+
+		KaleoInstance kaleoInstance =
+			_kaleoInstanceLocalService.addKaleoInstance(
+				1, 1, RandomTestUtil.randomString(), 1,
+				HashMapBuilder.<String, Serializable>put(
+					WorkflowConstants.CONTEXT_ENTRY_CLASS_NAME,
+					"com.liferay.blogs.model.BlogsEntry"
+				).put(
+					WorkflowConstants.CONTEXT_SERVICE_CONTEXT,
+					(Serializable)serviceContext
+				).build(),
+				serviceContext);
+
+		JSONObject workflowContextJSONObject = JSONFactoryUtil.createJSONObject(
+			kaleoInstance.getWorkflowContext());
+
+		JSONObject mapJSONObject = workflowContextJSONObject.getJSONObject(
+			"map");
+
+		mapJSONObject.remove("javaClass");
+
+		JSONObject serviceContextJSONObject = mapJSONObject.getJSONObject(
+			"serviceContext");
+
+		serviceContextJSONObject.put(
+			"javaClass",
+			"com.liferay.headless.common.spi.service.context." +
+				"ServiceContextUtil$1");
+
+		kaleoInstance.setWorkflowContext(workflowContextJSONObject.toString());
+
+		_kaleoInstance = _kaleoInstanceLocalService.updateKaleoInstance(
+			kaleoInstance);
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				_CLASS_NAME, LoggerTestUtil.OFF)) {
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, _CLASS_NAME);
+
+			upgradeProcess.upgrade();
+
+			_multiVMPool.clear();
+		}
+
+		KaleoInstance kaleoInstance =
+			_kaleoInstanceLocalService.fetchKaleoInstance(
+				_kaleoInstance.getKaleoInstanceId());
+
+		JSONObject workflowContextJSONObject = JSONFactoryUtil.createJSONObject(
+			kaleoInstance.getWorkflowContext());
+
+		JSONObject mapJSONObject = workflowContextJSONObject.getJSONObject(
+			"map");
+
+		JSONObject serviceContextJSONObject = mapJSONObject.getJSONObject(
+			"serviceContext");
+
+		Assert.assertNotEquals(
+			"com.liferay.headless.common.spi.service.context." +
+				"ServiceContextUtil$1",
+			serviceContextJSONObject.get("javaClass"));
```

Comment:
@JhossephG please change it to 

```java
Assert.assertEquals("com.liferay.portal.kernel.service.ServiceContext", serviceContextJSONObject.get("javaClass"))
``` 
to make sure the final value is the expected.

Please also add something like:

```java
Assert.assertTrue(serviceContextJSONObject.containsKey("serializable");
```